### PR TITLE
Manually create AWS state disk symlink

### PR DESCRIPTION
**Pull request by malt3**
### Proposed change(s)

- Manually create AWS state disk symlink

### Checklist

- [x] Wait for image https://github.com/edgelesssys/constellation/actions/runs/3310764256
- [x] Link to Milestone

### CODE REVIEW

**Comment by Nirusu** [comment]
1) Merge #355 for tail & tr 
2) Script gives a warning: 
```bash
warning: command substitution: ignored null byte in input
```
3) We cannot create a symlink from /dev/sdb to /dev/nvme1. It has to be /dev/sdb -> /dev/nvme1n1, otherwise the disk-mapper will crash immediately.

Tested on t3.micro with Ubuntu manually:
```bash
# Broken
AWS state disk found
/dev/sdb → /dev/nvme1
sudo ./disk-mapper --csp aws
{"level":"INFO","ts":"2022-10-21T17:53:53Z","caller":"cmd/main.go:58","msg":"Starting disk-mapper","version":"","cloudProvider":"aws"}
Device /dev/nvme1 is not compatible.
{"level":"FATAL","ts":"2022-10-21T17:53:53Z","caller":"cmd/main.go:121","msg":"Failed to initialize device mapper","error":"initializing crypt device for disk \"/dev/nvme1\": libcryptsetup function 'crypt_init' returned error with code '-15'."}
```

Changing this to nvme1n1:
```bash
# Working
AWS state disk found
/dev/sdb → /dev/nvme1n1
sudo ./disk-mapper --csp aws
{"level":"INFO","ts":"2022-10-21T17:54:45Z","caller":"cmd/main.go:58","msg":"Starting disk-mapper","version":"","cloudProvider":"aws"}
{"level":"INFO","ts":"2022-10-21T17:54:45Z","logger":"setupManager","caller":"setup/setup.go:117","msg":"Preparing new state disk"}
{"level":"INFO","ts":"2022-10-21T17:54:53Z","caller":"mapper/mapper.go:127","msg":"Wiping disk","progress":"0.00%"}
{"level":"INFO","ts":"2022-10-21T17:55:24Z","caller":"mapper/mapper.go:127","msg":"Wiping disk","progress":"47.71%"}
{"level":"INFO","ts":"2022-10-21T17:55:53Z","caller":"mapper/mapper.go:127","msg":"Wiping disk","progress":"93.03%"}
{"level":"INFO","ts":"2022-10-21T17:55:58Z","caller":"mapper/mapper.go:147","msg":"Wiping disk successful","duration":64.731132505}
```

Now I don't know if we can just assume both `nvmXnY` or have to enumerate through both. Note that existing symlinks will cause an error in the current script and abort. So if enumerating both is the solution, that needs to be ignored, too.

**Comment by daniel-weisse** [comment]
>Now I don't know if we can just assume both `nvmXn1` or have to enumerate through both variables `nvmXnY`

From [the AWS docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html), `Y` is always `1`

File: image/mkosi.skeleton/usr/lib/dracut/modules.d/39constellation-mount/aws-nvme-disk.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+set -euo pipefail
+shopt -s extglob nullglob
+
+AWS_STATE_DISK_DEVICENAME="sdb"
+AWS_STATE_DISK_SYMLINK="/dev/${AWS_STATE_DISK_DEVICENAME}"
+
+# hack: aws nvme udev rules are never executed. Create symlinks for the nvme devices manually.
+while [ ! -L "${AWS_STATE_DISK_SYMLINK}" ]
+do
+    for nvmedisk in /dev/nvme+([0-9])n1
```
Comment (Nirusu) [inline image/mkosi.skeleton/usr/lib/dracut/modules.d/39constellation-mount/aws-nvme-disk.sh:15]:
```suggestion
    for nvmedisk in /dev/nvme*n1
```

File: image/mkosi.skeleton/usr/lib/dracut/modules.d/39constellation-mount/aws-nvme-disk.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    if [ -L "${AWS_STATE_DISK_SYMLINK}" ]; then
+        break
+    fi
+    echo "Waiting for state disk to appear.."
+    sleep 2
+done
+
+echo "AWS state disk found"
+echo ${AWS_STATE_DISK_SYMLINK} → $(readlink -f "${AWS_STATE_DISK_SYMLINK}")
+sleep 2
```
Comment (Nirusu) [inline image/mkosi.skeleton/usr/lib/dracut/modules.d/39constellation-mount/aws-nvme-disk.sh:31]:
```suggestion
```
Unnecessarily stalls the boot.

**Pull request merged**